### PR TITLE
Provide non blocking API

### PR DESCRIPTION
**Pull request by rnestler**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -505,9 +502,13 @@ where
     /// Run a temperature/humidity measurement and return the combined result.
     ///
     /// This is a blocking function call.
-    pub fn measure(&mut self, mode: PowerMode) -> Result<Measurement, Error<E>> {
+    pub fn measure<D: DelayUs<u16>>(
+        &mut self,
+        mode: PowerMode,
+        delay: &mut D,
+    ) -> Result<Measurement, Error<E>> {
```
Comment (dbrgn) [inline src/lib.rs]: You could also use the impl trait syntax:

```suggestion
    pub fn measure(
        &mut self,
        mode: PowerMode,
        delay: &mut impl DelayUs<u16>,
    ) -> Result<Measurement, Error<E>> {
```

Might be easier to read, what do you think?

**Review by dbrgn — CHANGES_REQUESTED** [review]
Can you provide multiple impl blocks (common functionality, blocking API, non-blocking API) so we get nicer docs?

File: src/lib.rs (hunk 1 above)
Comment (rnestler) [reply src/lib.rs]: Fixed.

File: src/lib.rs [hunk 2]
```diff
@@ -465,6 +465,27 @@ where
...
+    /// internal state machines and reload calibration data from the memory.
+    pub fn reset(&mut self, delay: &mut impl DelayUs<u16>) -> Result<(), Error<E>> {
+        self.send_command(Command::SoftwareReset)?;
+        // Table 5: 180-240 µs
+        delay.delay_us(240);
+        Ok(())
+    }
+}
+
+/// Asynchronnous functions for measurement
```
Comment (dbrgn) [inline src/lib.rs]:
```suggestion
/// Asynchronous functions for starting / reading measurements.
```

File: src/lib.rs [hunk 3]
```diff
@@ -499,6 +515,42 @@ where
...
+
+    /// Read the result of a humidity measurement.
+    pub fn get_humidity_measurement_result(&mut self) -> Result<Humidity, Error<E>> {
+        let mut buf = [0; 3];
+        self.read_with_crc(&mut buf)?;
+        Ok(Humidity::from_raw(u16::from_be_bytes([buf[0], buf[1]])))
+    }
+}
+
+/// Blocking functions for measurement
```
Comment (dbrgn) [inline src/lib.rs]:
```suggestion
/// Blocking functions for doing measurements.
```

**Review by dbrgn — CHANGES_REQUESTED** [review]

**Review by dbrgn — APPROVED** [review]
Thanks! There are some more things that should be done before the next release:

- Provide an async wakeup function as well
- Add notes about async to README and to crate-level documentation
- Add an example that uses async calls (ideally an embedded one)

However all of them can be added in separate PRs, so let's merge this.

@rnestler can you squash the last two commits before merging?

**Pull request merged**